### PR TITLE
MOTOR-951 Test against MongoDB 6.0

### PR DESCRIPTION
**Pull request by juliusgeo**

### CODE REVIEW

**Comment by blink1073** [comment]
Needs a local pre-commit run for `black` and `isort`.

File: synchro/__init__.py [hunk 1]
```diff
@@ -51,6 +51,11 @@
     ssl_support,
     write_concern,
 )
+
+try:
```
Comment (blink1073) [inline synchro/__init__.py:55]: We should add a note here that we're importing for API compat with `pymongo`

File: test/asyncio_tests/test_asyncio_change_stream.py [hunk 2]
```diff
@@ -181,14 +182,16 @@ async def test_close(self):
...
+    @unittest.skip("Failing due to: https://jira.mongodb.org/browse/PYTHON-3389.")
     async def test_missing_id(self):
         coll = self.collection
         change_stream = coll.watch([{"$project": {"_id": 0}}])
         future = change_stream.next()
         self.wait_and_insert(change_stream)
         with self.assertRaises((InvalidOperation, OperationFailure)):
             await future
-
+        with self.assertRaises(OperationFailure):
```
Comment (blink1073) [inline test/asyncio_tests/test_asyncio_change_stream.py:193]: Is this still needed?

**Review by blink1073 — CHANGES_REQUESTED** [review]

**Review by blink1073 — APPROVED** [review]
LGTM!

**Review by ShaneHarvey — APPROVED** [review]
Nice!

**Pull request merged**